flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/MetastoreCommitPolicy.java

```diff
@@ -45,6 +45,9 @@ public void setMetastore(TableMetaStore metaStore) {
 	public void commit(Context context) throws Exception {
 		LinkedHashMap<String, String> partitionSpec = context.partitionSpec();
 		metaStore.createOrAlterPartition(partitionSpec, context.partitionPath());
+		if (metaStore.getPartition(partitionSpec).isPresent()) {
```

Comment:
NIT: print previous path too, you can do this:
```
		metaStore.getPartition(partitionSpec).ifPresent(path ->
				LOG.warn("The partition {} has existed before current commit," +
						" the path is {}, this partition will be altered instead of being created",
						partitionSpec, path));
```

Reply:
Sure, code has been changed following ur suggestion